src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,80 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final String LS = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] totals = readAndAggregate(fromFileName);
+        String report = buildReport(totals[0], totals[1]);
+        writeReport(toFileName, report);
+    }
+
+    private int[] readAndAggregate(String fromFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try (BufferedReader br = Files.newBufferedReader(
+                Paths.get(fromFileName), StandardCharsets.UTF_8)) {
+            String line;
+            int lineNumber = 0;
+            while ((line = br.readLine()) != null) {
+                lineNumber++;
+                String trimmedLine = line.trim();
+                if (trimmedLine.isEmpty()) {
+                    continue;
+                }
+                String[] parts = trimmedLine.split(DELIMITER);
+                if (parts.length != 2) {
+                    continue;
```

This method currently continues when a line doesn't split into exactly two parts. The implementation must fail fast on invalid format. This violates: "The parser should correctly handle each line by splitting on a single comma into exactly two parts: operation and amount." Instead of `continue`, throw a RuntimeException that includes the file name and line number so malformed input is not silently ignored.